src/NServiceBus.Core/Diagnostics/ActivityExtensions.cs

```diff
@@ -21,7 +27,8 @@ public static bool TryGetRecordingPipelineActivity(this ContextBag pipelineConte
         return false;
     }
 
-    public static void SetPipelineActitvity(this ContextBag pipelineContext, Activity activity) => pipelineContext.Set(OutgoingActivityKey, activity);
+    public static void SetOutgoingPipelineActitvity(this ContextBag pipelineContext, Activity activity) => pipelineContext.Set(OutgoingActivityKey, activity);
```

Comment:
Consider retargeting these to `IIncomingContext` and `IOutgoingContext` to prevent them being called from the wrong places.

Reply:
good point but there are some limitations. e.g. the subscribe/unsubscribe contexts are not considered outgoing contexts. And with this API technically it would be possible to add something to the receive-activity from an outgoing behavior, which I'm not sure would be valid or should be prevented.